crypto/ts/ts_rsp_sign.c

```diff
@@ -681,20 +669,20 @@ static int ts_RESP_sign(TS_RESP_CTX *ctx)
     certs = ctx->flags & TS_ESS_CERT_ID_CHAIN ? ctx->certs : NULL;
     if (ctx->ess_cert_id_digest == NULL
         || ctx->ess_cert_id_digest == EVP_sha1()) {
-        if ((sc = ess_SIGNING_CERT_new_init(ctx->signer_cert, certs)) == NULL)
+        if ((sc = ESS_SIGNING_CERT_new_init(ctx->signer_cert, certs, 0)) == NULL)
             goto err;
 
-        if (!ess_add_signing_cert(si, sc)) {
+        if (!ESS_SIGNING_CERT_add(si, sc)) {
             TSerr(TS_F_TS_RESP_SIGN, TS_R_ESS_ADD_SIGNING_CERT_ERROR);
             goto err;
         }
     } else {
-        sc2 = ess_signing_cert_v2_new_init(ctx->ess_cert_id_digest,
-                                           ctx->signer_cert, certs);
+        sc2 = ESS_SIGNING_CERT_V2_new_init(ctx->ess_cert_id_digest,
+                                           ctx->signer_cert, certs, 0);
         if (sc2 == NULL)
             goto err;
 
-        if (!ess_add_signing_cert_v2(si, sc2)) {
+        if (!ESS_SIGNING_CERT_V2_add(si, sc2)) {
```

Comment:
[3/3] Consequently, in `ts_RESP_sign()` the signing certificate is now inserted using `CMS_signed_add1_attr_by_NID(...)` instead of `PKCS7_add_signed_attribute()`, right?

Reply:
@mspncp : boulette ;)

Reply:
Boulette? You mean the meat ball?

Reply:
Ah, I see. it also means [a mistake](https://defr.dict.cc/?s=boulette).

Reply:
Well, it's not mistake if those two functions do the same thing. But then maybe they should be based on a common implementation?

Reply:
There is very few things to factorize here, if I remember well .
I don't see/understand why this SignerInfo swapping is compiling.

Reply:
no @mspncp ts_RESP_sign use PKCS7_add_signed_attribute() (look ESS_SIGNING_CERT_V2_add at crypto/ess/ess_lib.c) as well as all the TS code uses the PKCS7 functions instead CMS functions.
I have not touched anything of the TS code (I just moved to crypto/ess ESS functions ;)